### PR TITLE
[stable/node-problem-detector] Split metrics enable flag from service monitor

### PR DESCRIPTION
**Pull request by spirosoik**
## Description

The reason of decoupling service monitor from service is that sometimes
you want just to export the service and in your prometheus configuration
(prometheus-operator) define the service monitor for this service.

So it's better to have a separate flag for exposing the service instead of 
coupling it with service monitor.

## Checklist

- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)
- [x] I have read the [contribution instructions](https://github.com/deliveryhero/helm-charts#contributing), bumped chart version and regenerated the docs
- [x] Github actions are passing

### CODE REVIEW

**Review by max-rocket-internet — APPROVED** [review]
Nice, thanks @spirosoik!

**Comment by spirosoik** [comment]
@max-rocket-internet fixed the broken check all docs, so please when you can have a quick look again.

**Comment by spirosoik** [comment]
@max-rocket-internet ping

File: stable/node-problem-detector/README.md [hunk 1]
```diff
@@ -1,42 +1,16 @@
 # node-problem-detector
 
-![Version: 2.0.4](https://img.shields.io/badge/Version-2.0.4-informational?style=flat-square) ![AppVersion: v0.8.9](https://img.shields.io/badge/AppVersion-v0.8.9-informational?style=flat-square)
+![Version: 2.0.5](https://img.shields.io/badge/Version-2.0.5-informational?style=flat-square) ![AppVersion: v0.8.9](https://img.shields.io/badge/AppVersion-v0.8.9-informational?style=flat-square)
 
 This chart installs a [node-problem-detector](https://github.com/kubernetes/node-problem-detector) daemonset. This tool aims to make various node problems visible to the upstream layers in cluster management stack. It is a daemon which runs on each node, detects node problems and reports them to apiserver.
 
 **Homepage:** <https://github.com/kubernetes/node-problem-detector>
 
-## How to install this chart
```
Comment (max-rocket-internet) [inline stable/node-problem-detector/README.md:9]: Why is all this being removed?

**Review by max-rocket-internet — CHANGES_REQUESTED** [review]

**Comment by max-rocket-internet** [comment]
> @max-rocket-internet ping

Sorry, I was on holiday

File: stable/node-problem-detector/README.md (hunk 1 above)
Comment (spirosoik) [reply stable/node-problem-detector/README.md:9]: let me check

Comment (spirosoik) [reply stable/node-problem-detector/README.md:9]: I think the auto-generated readme removed it.

Comment (spirosoik) [reply stable/node-problem-detector/README.md:9]: let me revert back

Comment (spirosoik) [reply stable/node-problem-detector/README.md:9]: Though if I won't use helm-docs then it breaks the `Check all docs` pipeline

Comment (max-rocket-internet) [reply stable/node-problem-detector/README.md:9]: This text needs to stay. Are you running `helm-docs` from repo root?

Comment (spirosoik) [reply stable/node-problem-detector/README.md:9]: let me give it one more try

Comment (spirosoik) [reply stable/node-problem-detector/README.md]: done

**Comment by spirosoik** [comment]
@max-rocket-internet should be fine now

**Comment by spirosoik** [comment]
@max-rocket-internet all good

**Review by max-rocket-internet — APPROVED** [review]

**Pull request merged**